apps/dev-api/domains/miniapp/schema/OIDCClient.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Generated by `createschema miniapp.OIDCClient 'b2cApp:Relationship:B2CApp:CASCADE; clientId:Text; clientSecret:Text; grantTypes:Json; responseTypes:Json; developmentRedirectUri:Url; productionRedirectUri:Url'`
+ */
+
+const { faker } = require('@faker-js/faker')
+
+const { historical, versioned, uuided, tracked, softDeleted, dvAndSender } = require('@open-condo/keystone/plugins')
+const { GQLListSchema } = require('@open-condo/keystone/schema')
+
+const access = require('@dev-api/domains/miniapp/access/OIDCClient')
+const { OIDC_CLIENT_UNIQUE_B2C_APP_CONSTRAINT } = require('@dev-api/domains/miniapp/constants/constraints')
+const { OIDC_SECRET_LENGTH, OIDC_SECRET_CHAR_PATTERN } = require('@dev-api/domains/miniapp/constants/oidc')
+const { exportable } = require('@dev-api/domains/miniapp/plugins/exportable')
+const { GRANT_TYPES_FIELD, RESPONSE_TYPES_FIELD, TOKEN_AUTH_METHOD_FIELD } = require('@dev-api/domains/miniapp/schema/fields/OIDCClient')
+
+
+const OIDCClient = new GQLListSchema('OIDCClient', {
+    schemaDoc: 'OIDC client configuration for a specific app',
+    fields: {
+        b2cApp: {
+            schemaDoc: 'Link to the B2C application to which this configuration applies',
+            type: 'Relationship',
+            ref: 'B2CApp',
+            isRequired: true,
+            knexOptions: { isNotNullable: true }, // Required relationship only!
+            kmigratorOptions: { null: false, on_delete: 'models.CASCADE' },
+            access: {
+                read: true,
+                create: true,
+                update: false,
+            },
+        },
+        clientId: {
+            schemaDoc: 'OIDC Client ID',
+            type: 'Virtual',
+            graphQLReturnType: 'ID',
+            resolver (item) {
+                return item.b2cApp
+            },
+            access: {
+                read: true,
+                create: false,
+                update: false,
+            },
+        },
+        clientSecret: {
+            schemaDoc: 'OIDC Client Secret',
+            type: 'Text',
+            isRequired: true,
+            hooks: {
+                resolveInput ({ operation, originalInput, fieldPath }) {
+                    // NOTE: Secret must be strong enough, so we should generate it server-side
+                    if (operation === 'create' || originalInput.hasOwnProperty(fieldPath)) {
+                        return faker.internet.password(OIDC_SECRET_LENGTH, false, OIDC_SECRET_CHAR_PATTERN)
```

Comment:
it's better to use crypto! for client secret generation!

Reply:
SECURITY ISSUE

Reply:
Rewritten with manual generation with `crypto` and `randomInt`. (Faker was using similar strategy btw)